pkg/controller/controller.go

```diff
@@ -276,13 +274,29 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 		return fmt.Errorf("invalid CSI PV")
 	}
 	volumeId := p.volumeHandleToId(volume.Spec.CSI.VolumeHandle)
+
 	req := csi.DeleteVolumeRequest{
 		VolumeId: volumeId,
 	}
+	// get secrets if StorageClass specifies it
+	storageClassName := volume.Spec.StorageClassName
+	if len(storageClassName) != 0 {
+		if storageClass, err := p.client.StorageV1().StorageClasses().Get(storageClassName, metav1.GetOptions{}); err == nil {
+			if secretName, ok := storageClass.Parameters[secretNameKey]; ok {
```

Comment:
Please refactor a function that can be shared in `Provision()` and `Delete()` calls.

Reply:
done, PTAL.